crates/services/upgradable-executor/src/instance.rs

```diff
@@ -176,7 +190,64 @@ impl Instance<Created> {
             };
 
             let txs: Vec<_> = source
-                .next(gas_limit)
+                .next(gas_limit, u16::MAX)
+                .into_iter()
+                .map(|tx| match tx {
+                    MaybeCheckedTransaction::CheckedTransaction(checked, _) => {
+                        let checked: Checked<Transaction> = checked.into();
+                        let (tx, _) = checked.into();
+                        tx
+                    }
+                    MaybeCheckedTransaction::Transaction(tx) => tx,
+                })
+                .collect();
+
+            let encoded_txs = postcard::to_allocvec(&txs).map_err(|e| {
+                ExecutorError::Other(format!(
+                    "Failed encoding of the transactions for `peek_next_txs_size` function: {}",
+                    e
+                ))
+            })?;
+            let encoded_size = u32::try_from(encoded_txs.len()).map_err(|e| {
+                ExecutorError::Other(format!(
+                    "The encoded transactions are more than `u32::MAX`. We support only wasm32: {}",
+                    e
+                ))
+            })?;
+
+            caller
+                .data_mut()
+                .next_transactions
+                .entry(encoded_size)
+                .or_default()
+                .push(encoded_txs);
+            Ok(encoded_size)
+        };
+
+        Func::wrap(&mut self.store, closure)
+    }
+
+    fn peek_next_txs_size_with_limit<TxSource>(
```

Comment:
Ideally I would like to factor the closure to a dedicated function, and call it both from `peek_next_txs_size` and `peek_next_txs_size_with_limits`, but this causes the type of the closure to become `FnOnce` (instead of `Fn`), and `Func::wrap` cannot be applied to it.